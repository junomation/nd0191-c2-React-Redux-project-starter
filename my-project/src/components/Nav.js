import React from 'react'
import { useNavigate } from 'react-router-dom'

const Nav = () => {
  const navigate = useNavigate();

  const handleNavigate = (e, path) => {
    e.preventDefault();
    navigate(path);
  }

  return (
    <nav className='nav'>
      <ul>
        <li>
          <a href='/' className='active' onClick={(e) => handleNavigate(e, '/')}>
            Home
          </a>
        </li>
        <li>
          <a href='/add' onClick={(e) => handleNavigate(e, '/add')}>
            New Question
          </a>
        </li>
        <li>
          <a href='/leaderboard' onClick={(e) => handleNavigate(e, '/leaderboard')}>
            Leaderboard
          </a>
        </li>
      </ul>
    </nav>
  )
}

export default Nav