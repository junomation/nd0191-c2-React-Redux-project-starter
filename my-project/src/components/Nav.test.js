import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Nav from './Nav';
import { MemoryRouter as Router } from 'react-router-dom';

const mockAuthedUser = 'user1';
const mockUsers = {
  user1: {
    id: 'user1',
    name: 'John Doe',
    avatarURL: 'https://www.example.com/avatar1.png',
  },
  user2: {
    id: 'user2',
    name: 'Jane Smith',
    avatarURL: 'https://www.example.com/avatar2.png',
  },
};

const mockStore = {
  getState: () => ({ authedUser: mockAuthedUser, users: mockUsers }),
  subscribe: () => {},
  dispatch: () => {},
};

describe('Nav component', () => {
  test('renders correctly with mocked data', () => {
    const { container } = render(
      <Router>
        <Provider store={mockStore}>
          <Nav />
        </Provider>
      </Router>
    );

    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });
});
