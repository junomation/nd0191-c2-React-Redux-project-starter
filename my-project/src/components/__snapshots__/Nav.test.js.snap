// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders correctly with mocked data 1`] = `
<nav
  class="nav"
>
  <ul>
    <li>
      <a
        class="active"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class="inactive"
        href="/add"
      >
        New Question
      </a>
    </li>
    <li>
      <a
        class="inactive"
        href="/leaderboard"
      >
        Leaderboard
      </a>
    </li>
    <li
      class="authed-user"
    >
      <span
        class="nav-welcome"
      >
        Hello, 
        John Doe
      </span>
      <button
        class="nav-logout"
      >
        Logout
      </button>
    </li>
  </ul>
</nav>
`;
